### PR TITLE
VideoPress: Filter files by extension on selection

### LINKED ISSUE #26355: VideoPress: Filter files by mimetype
The file uploader needs to filter out non-video files both during file selection (with the `accept` parameter) and after files have been dragged over the `VideoUploadArea` component.

refs: https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/accept

### PR DESCRIPTION
**Pull request by dhasilva**
Closes #26355

#### Changes proposed in this Pull Request:

* Adds extension-based filtering for video file selector

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Open Storybook: `cd projects/js-packages/storybook && pnpm run storybook:dev`
* Navigate to Packages > VideoPress > VideoUploadArea
* Click `Select file to upload`
* Verify that only files with certain extensions are shown

### CODE REVIEW

File: projects/packages/videopress/src/client/state/constants.js [hunk 1]
```diff
@@ -24,6 +24,11 @@ export const REMOVE_VIDEO = 'REMOVE_VIDEO';
 export const SET_IS_FETCHING_UPLOADED_VIDEO_COUNT = 'SET_IS_FETCHING_UPLOADED_VIDEO_COUNT';
 export const SET_UPLOADED_VIDEO_COUNT = 'SET_UPLOADED_VIDEO_COUNT';
 
+/*
+ * Accepted file extensions
+ */
+export const VIDEO_EXTENSIONS = [ 'ogv', 'mp4', 'mov', 'wmv', 'avi', 'mpg', '3gp', '3g2', 'm4v' ];
```
Comment (renatoagds) [inline projects/packages/videopress/src/client/state/constants.js:30]: Don't think this is the place to put this value. IMO it should exists inside the `video-upload-area`component

**Comment by nunyvega** [comment]
@dhasilva I followed the steps and it tests well! I'm not approving it yet so Renato's notes can be addressed

File: projects/packages/videopress/src/client/state/constants.js (hunk 1 above)
Comment (retrofox) [reply projects/packages/videopress/src/client/state/constants.js:30]: The site should define these values and not be hard coded. Maybe we can create a follow-up PR to expose the mime types to the client.

Comment (retrofox) [reply projects/packages/videopress/src/client/state/constants.js:30]: created an issue https://github.com/Automattic/jetpack/issues/26468

**Review by retrofox — APPROVED** [review]

File: projects/packages/videopress/src/client/state/constants.js (hunk 1 above)
Comment (renatoagds) [reply projects/packages/videopress/src/client/state/constants.js:30]: This should not be the place to add this mime for now

**Pull request merged**